pkg/controllers/sync/controller.go

```diff
@@ -1096,16 +1096,17 @@ func (s *SyncController) removeClusterFinalizer(ctx context.Context, cluster *fe
 
 func (s *SyncController) reconcileCluster(qualifiedName common.QualifiedName) worker.Result {
 	cluster, found, err := s.informer.GetCluster(qualifiedName.Name)
-	logger := s.logger.WithValues("cluster-name", cluster.Name)
-	ctx := klog.NewContext(context.TODO(), logger)
 	if err != nil {
-		logger.Error(err, "Failed to get federated cluster")
+		s.logger.Error(err, "Failed to get federated cluster")
 		return worker.StatusError
 	}
 	if !found {
 		return worker.StatusAllOK
 	}
 
+	logger := s.logger.WithValues("cluster-name", cluster.Name)
```

Comment:
```suggestion
	logger := s.logger.WithValues("cluster-name", qualifiedName.String())
```
This line should proceed the first logging statement.